src/olympia/devhub/tasks.py

```diff
@@ -619,6 +620,72 @@ def track_validation_stats(json_result, addons_linter=False):
                 .format(runner, listed_tag, result_kind))
 
 
+@task
+@write
+@set_modified_on
+def pngcrush_existing_icons(addon_id):
+    """
+    Call pngcrush_image() on the icons of a given add-on.
+    """
+    log.info('Crushing icons for add-on %s', addon_id)
+    addon = Addon.objects.get(pk=addon_id)
+    if addon.icon_type != 'image/png':
+        log.info('Aborting icon crush for add-on %s, icon type is not a PNG.',
+                 addon_id)
+        return
+    icon_dir = addon.get_icon_dir()
+    pngcrush_image(os.path.join(icon_dir, '%s-64.png' % addon_id))
+    pngcrush_image(os.path.join(icon_dir, '%s-32.png' % addon_id))
+    # Return an icon hash that set_modified_on decorator will set on the add-on
+    # after a small delay. This is normally done with the true md5 hash of the
+    # original icon, but we don't necessarily have it here. We could read one
+    # of the icons we modified but it does not matter just fake a hash to
+    # indicate it was "manually" crushed.
+    return {
+        'icon_hash': 'mcrushed'
```

Comment:
Maybe use a unix timestamp? I'm not sure of any implications but that seems the most reasonable way after an md5 hash